label-application-rule-config-service-api/src/main/proto/org/hypertrace/label/application/rule/config/service/v1/label_application_rule.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+package org.hypertrace.label.application.rule.config.service.v1;
+
+option java_multiple_files = true;
+
+message LabelApplicationRule {
+  string id = 1;
+  LabelApplicationRuleData data = 2;
+}
+
+message LabelApplicationRuleData {
+  string name = 1;
+  MatchingCriteria matching_criteria = 2;
+  ApplicationCriteria application_criteria = 3;
+
+  message MatchingCriteria {
+    Condition condition = 1;
+
+    message Condition {
+      oneof condition {
+        LeafCondition leaf_condition = 1;
+        CompositeCondition composite_condition = 2;
+      }
+    }
+
+    message CompositeCondition {
+      LogicalOperator operator = 1;
+      repeated Condition children = 2;
+
+      enum LogicalOperator {
+        LOGICAL_OPERATOR_UNSPECIFIED = 0;
+        LOGICAL_OPERATOR_AND = 1;
+        LOGICAL_OPERATOR_OR = 2;
+      }
+    }
+
+    message LeafCondition {
+      Scope scope = 1;
+      string key = 2;
+      oneof condition {
+        StringCondition string_condition = 3;
+        UnaryCondition unary_condition = 4;
+        JsonCondition json_condition = 5;
+      }
+    }
+
+    message StringCondition {
+      Operator operator = 1;
+      string value = 2;
+
+      enum Operator {
+        OPERATOR_UNSPECIFIED = 0;
+        OPERATOR_EQUALS = 1; // operator to check if the key exists, and value is equal to the provided value
+        OPERATOR_MATCHES_REGEX = 2; // operator to check if the key exists, and value matches the provided regex value
+      }
+    }
+
+    message UnaryCondition {
+      // used to check the existence of the key
```

Comment:
may want to still have an operator here as there may be multiple unary conditions in the future (e.g. exists and not exists)

Reply:
Done